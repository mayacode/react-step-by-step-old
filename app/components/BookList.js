import React from 'react';
import { shape, string, arrayOf } from 'prop-types';

import BookListItem from './BookListItem';

const BookList = ({ books }) => <div>{books.map(book => <BookListItem
  key={book.id}
  title={book.volumeInfo.title}
/>)}</div>;

BookList.displayName = 'BookList';
BookList.propTypes = {
  books: arrayOf(shape({
    volumeInfo: shape({
      title: string,
    }),
    id: string,
  })).isRequired,
};

export default BookList;
