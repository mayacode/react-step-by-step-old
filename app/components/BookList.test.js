import React from 'react';
import { shallow } from 'enzyme';

import BookList from './BookList';

describe('<BookList />', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<BookList books={[]} />);
        expect(wrapper).toMatchSnapshot();
    });
});