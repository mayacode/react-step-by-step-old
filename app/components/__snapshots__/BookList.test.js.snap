// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookList /> matches a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookList
    books={
      Array [
        Object {
          "id": "1234qwer",
          "volumeInfo": Object {
            "title": "A Book",
          },
        },
        Object {
          "id": "1235qwer",
          "volumeInfo": Object {
            "title": "A Book 2",
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BookListItem
          title="A Book"
        />,
        <BookListItem
          title="A Book 2"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1234qwer",
        "nodeType": "function",
        "props": Object {
          "title": "A Book",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1235qwer",
        "nodeType": "function",
        "props": Object {
          "title": "A Book 2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BookListItem
            title="A Book"
          />,
          <BookListItem
            title="A Book 2"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1234qwer",
          "nodeType": "function",
          "props": Object {
            "title": "A Book",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1235qwer",
          "nodeType": "function",
          "props": Object {
            "title": "A Book 2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
